Formula/cloudflare-wrangler.rb

```diff
@@ -0,0 +1,17 @@
+class CloudflareWrangler < Formula
+  desc "CLI tool for Cloudflare Workers"
+  homepage "https://github.com/cloudflare/wrangler"
+  url "https://github.com/cloudflare/wrangler/archive/v1.13.0.tar.gz"
+  sha256 "078ca2e7941f6476c18594fa7e7d26da35dfae2b65557d3ad6d598844cf915b4"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_equal "👷 ✨  wrangler #{version}\n", shell_output("#{bin}/wrangler --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
```
USAGE:
    wrangler [SUBCOMMAND]

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

SUBCOMMANDS:
    kv:namespace    🗂️  Interact with your Workers KV Namespaces
    kv:key          🔑  Individually manage Workers KV key-value pairs
    kv:bulk         💪  Interact with multiple Workers KV key-value pairs at once
    route           ➡️  List or delete worker routes.
    secret          🤫  Generate a secret that can be referenced in the worker script
    generate        👯  Generate a new worker project
    init            📥  Create a wrangler.toml for an existing project
    build           🦀  Build your worker
    preview         🔬  Preview your code temporarily on cloudflareworkers.com
    dev             👂  Start a local server for developing your worker
    publish         🆙  Publish your worker to the orange cloud
    config          🕵️  Authenticate Wrangler with a Cloudflare API Token or Global API Key
    subdomain       👷  Configure your workers.dev subdomain
    whoami          🕵️  Retrieve your user info and test your auth config
    tail            🦚  Aggregate logs from production worker
    login           🔓 Authenticate Wrangler with your Cloudflare username and password
    help            Prints this message or the help of the given subcommand(s)
```

Most subcommands require a valid Cloudflare account to work properly, with only 3 exceptions. IMHO, none of these 3 subcommands suites the test purpose.

## `init`

create `wrangler.toml` from a template
```
name = "name-specified-on-cmdline"
type = "javascript/webpack/rust"
account_id = ""
workers_dev = true
route = ""
zone_id = ""
```

## `generate`

clone https://github.com/cloudflare/worker-template into a directory, and `wrangler init` inside

## `build`

Does nothing for `type=javascript`. Call external programs to build project for `type=webpack/rust`.

Reply:
> Most subcommands require a valid Cloudflare account to work properly

From my message:

> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.

Reply:
resolved, PTAL